apis/kueue/v1alpha1/dynamicresourceallocationconfig_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+
+// DynamicResourceAllocationConfig is a singleton CRD that maps a logical resource name to one or more DeviceClasses
+// in the cluster. Only one instance named "default" in the kueue-system namespace
+// is allowed.
+type DynamicResourceAllocationConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec DynamicResourceAllocationConfigSpec `json:"spec"`
+}
+
+// DynamicResourceAllocationConfigSpec holds all resource to DeviceClass mappings.
+type DynamicResourceAllocationConfigSpec struct {
+	// Resources lists logical resources that Kueue will account.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	Resources []DynamicResource `json:"resources"`
+}
+
+// DynamicResource describes a single logical resource and the DeviceClasses mapping. The resource name is used
+// to quota in ClusterQueue.
+type DynamicResource struct {
+	// Name is referenced in ClusterQueue.nominalQuota and Workload status.
```

Comment:
Is there any validation we need on a name?

dns rules?

length?

Reply:
it is covered in the webhook halidation, this name basically needs to be a fully qualified name.https://github.com/alaypatel07/kueue/blob/68b32feae4c16de3c2d627a402997eace515f4f5/pkg/webhooks/dynamicresourceallocationconfig_webhook.go#L111

Reply:
It would be worth having as a comment.

Ideally you could do this via cel and avoid the code in the webhook.

Reply:
you can resolve this.